core/app/services/uffizzi_core/deployment_service.rb

```diff
@@ -12,12 +12,13 @@ class UffizziCore::DeploymentService
   }.freeze
 
   class << self
-    def create_from_compose(compose_file, project, user)
+    def create_from_compose(compose_file, project, user, metadata = {})
       deployment_attributes = ActionController::Parameters.new(compose_file.template.payload)
       deployment_form = UffizziCore::Api::Cli::V1::Deployment::CreateForm.new(deployment_attributes)
       deployment_form.assign_dependences!(project, user)
       deployment_form.compose_file = compose_file
       deployment_form.creation_source = UffizziCore::Deployment.creation_source.compose_file_manual
+      deployment_form.metadata = metadata || {}
```

Comment:
I don't think we need `metadata || {}`. `{}` is the default for metadata in this method.